### PR TITLE
new: Add support for Linode Disk Encryption

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This pull request adds support for Linode Disk Encryption, including the following changes:

* New `InstanceDiskEncryptionType` enum
* New `disk_encryption` argument for the LinodeGroup(...).instance_create(...) and Instance(...).rebuild(...) methods
* New `disk_encryption` and `lke_cluster_id` attributes under the Instance class
* New `lke_cluster` property method under the Instance class
* New `disk_encryption` attribute under the Disk class
* New `disk_encryption` attribute under the LKENodePool class

## ✔️ How to Test

The following test steps assume you have pulled down this PR locally and run `make install`. Additionally, your environment should be configured to point to an API instance where LDE is available:

```
export LINODE_API_URL=https://.../v4beta
export LINODE_API_CA=$PWD/cacert.pem
export LINODE_TOKEN=...
```

### Unit Testing

```
make testunit
```

### Integration Testing

```
make TEST_COMMAND=models/lketestint
make TEST_COMMAND=models/linode testint
```

#### Instance-related Changes

1. In a Python SDK sandbox environment (e.g. dx-devenv), run the following:

```python
import os

from linode_api4 import LinodeClient, InstanceDiskEncryptionType

client = LinodeClient(
    base_url=os.getenv("LINODE_API_URL"),
    ca_path=os.getenv("LINODE_API_CA"),
    token=os.getenv("LINODE_TOKEN"),
)

target_region = [v for v in client.regions() if "Disk Encryption" in v.capabilities][0]

instance, _ = client.linode.instance_create(
    "g6-nanode-1",
    target_region,
    image="linode/alpine3.19",
    booted=False,
    disk_encryption=InstanceDiskEncryptionType.enabled,
)

print("Instance Disk Encryption:", instance.disk_encryption)

client.polling.wait_for_entity_free("linode", instance.id)

instance, _ = instance.rebuild(
    image="linode/alpine3.19",
    booted=False,
    disk_encryption=InstanceDiskEncryptionType.disabled,
)

print("Rebuilt Instance Disk Encryption:", instance.disk_encryption)
```

2. Ensure the output matches the following:

```
Instance Disk Encryption: enabled
Rebuilt Instance Disk Encryption: disabled
```

### LKE-related Changes

1. In a Python SDK sandbox environment (e.g. dx-devenv), run the following:

```python
import os
import time

from linode_api4 import LinodeClient, InstanceDiskEncryptionType, Instance

client = LinodeClient(
    base_url=os.getenv("LINODE_API_URL"),
    ca_path=os.getenv("LINODE_API_CA"),
    token=os.getenv("LINODE_TOKEN"),
)

target_region = [
    v
    for v in client.regions()
    if set(v.capabilities).issuperset({"Kubernetes", "Disk Encryption"})
][0]

cluster = client.lke.cluster_create(
    target_region,
    "test-cluster",
    node_pools=client.lke.node_pool("g6-standard-1", 1),
    kube_version="1.29"
)

pool = cluster.pools[0]

print("Pool Disk Encryption:", pool.disk_encryption)

node_instance_id = None

# Wait for a node to have a corresponding instance ID
while node_instance_id is None:
    pool.invalidate()
    node_instance_id = pool.nodes[0].instance_id
    time.sleep(5)

inst = Instance(client, node_instance_id)

print("Underlying Node Instance:", inst)
print("LKE Cluster Backref:", inst.lke_cluster)
```

2. Ensure the output looks similar to the following:

```
Pool Disk Encryption: enabled
Underlying Node Instance: Instance: 25182862
LKE Cluster Backref: LKECluster: 7582
```

### CODE REVIEW

File: linode_api4/objects/serializable.py [hunk 1]
```diff
@@ -223,3 +224,22 @@ def __delitem__(self, key):
 
     def __len__(self):
         return len(vars(self))
+
+
+class StrEnum(str, Enum):
```
Comment (lgarber-akamai) [inline linode_api4/objects/serializable.py:229]: This was originally implemented in the VM Placement project branch.

File: test/integration/helpers.py [hunk 2]
```diff
@@ -79,12 +79,14 @@ def wait_for_condition(
 
 
 # Retry function to help in case of requests sending too quickly before instance is ready
-def retry_sending_request(retries: int, condition: Callable, *args) -> object:
+def retry_sending_request(
+    retries: int, condition: Callable, *args, **kwargs
```
Comment (lgarber-akamai) [inline test/integration/helpers.py:83]: Added **kwargs here so I could specify `disk_encryption` in the rebuild arguments 👍

**Review by ykim-1 — APPROVED** [review]
LGTM, verified both manual steps and tests locally

**Review by zliang-akamai — APPROVED** [review]
Looks great, tests passed. Nice work!

File: linode_api4/objects/serializable.py [hunk 3]
```diff
@@ -223,3 +224,22 @@ def __delitem__(self, key):
...
+    """
+    Used for enums that are of type string, which is necessary
+    for implicit JSON serialization.
+
+    NOTE: Replace this with StrEnum once Python 3.10 has been EOL'd.
+    See: https://docs.python.org/3/library/enum.html#enum.StrEnum
```
Comment (zliang-akamai) [inline linode_api4/objects/serializable.py:235]: Lots of feature are not in older Python versions, and there are still 2 years before 3.10 retired. Let's just wait...

**Pull request merged**